locator/topology.cc

```diff
@@ -396,7 +397,7 @@ node_holder topology::pop_node(const node* node) {
 
     // shrink _nodes if the last node is popped
     // like when failing to index a newly added node
-    if (node->idx() == _nodes.size() - 1) {
+    if (std::cmp_equal(node->idx(), _nodes.size() - 1)) {
```

Comment:
Here I think it is safe to cast `node->idx()` manually to `size_t()`: `size_t(node->idx())`.

Reply:
i am not sure. unless `node::idx_type` is unsigned. BTW, i'd prefer `static_cast` instead of C style cast.